### ISSUE TITLE
Unexpected binaryTargets environment variables usage

### ISSUE THREAD

**Issue by eddeee888** [prisma/prisma#8112]
### Bug description

In version 2.22 and below, I was able to use an environment variable to set `binaryTargets`. This was the `generator` part in my schema:

```
generator client {
  provider          = "prisma-client-js"
  binaryTargets = [env("PRISMA_BINARY_TARGET")]
}
```

I could run something like this to set this environment variable:

```bash
$ PRISMA_BINARY_TARGET='native' yarn prisma generate
```

However, it does not work like this anymore in v2.26. If I upgrade Prisma to 2.26 and try running the same command, I'd get an error like this ( with `DEBUG="prisma*"` turned on):

![Screen Shot 2021-07-06 at 8 55 43 pm](https://user-images.githubusercontent.com/33769523/124588812-928e5d00-de9c-11eb-9941-9a233143ac7e.png)

### How to reproduce

1. Create a new schema
2. Use an environment variable for `binaryTargets`
3. Set the environment variable when running relevant commands like `prisma generate`

### Expected behavior

I was expecting v2.26 to work the same way as v2.22 and before:

```
$ PRISMA_BINARY_TARGET='native' yarn prisma generate
```

### Alternative

Given the following setting:

```
generator client {
  provider          = "prisma-client-js"
  binaryTargets = env("PRISMA_BINARY_TARGET") // Notice env is now the `binaryTargets` array 
}
```

Maybe the targets can be passed in with a `,` separator? For example:

```
$ PRISMA_BINARY_TARGET='native,darwin' yarn prisma generate
```

### Current workaround
After digging around a bit [in this file](https://github.com/prisma/prisma/blob/master/src/packages/sdk/src/utils/parseEnvValue.ts), I found that it was expecting an array in *JSON string* instead of a string.
That means I could do something like this:

```bash
$ PRISMA_BINARY_TARGET='[\"native\"]' yarn prisma generate
```

### Prisma information

N/A

### Environment & setup

- OS: MacOS
- Database: MySQL
- Node.js version: 12.13.1

### Prisma Version

```
prisma               : 2.26.0
@prisma/client       : 2.26.0
Current platform     : darwin
Query Engine         : query-engine 9b816b3aa13cc270074f172f30d6eda8a8ce867d (at node_modules/@prisma/engines/query-engine-darwin)
Migration Engine     : migration-engine-cli 9b816b3aa13cc270074f172f30d6eda8a8ce867d (at node_modules/@prisma/engines/migration-engine-darwin)
Introspection Engine : introspection-core 9b816b3aa13cc270074f172f30d6eda8a8ce867d (at node_modules/@prisma/engines/introspection-engine-darwin)
Format Binary        : prisma-fmt 9b816b3aa13cc270074f172f30d6eda8a8ce867d (at node_modules/@prisma/engines/prisma-fmt-darwin)
Default Engines Hash : 9b816b3aa13cc270074f172f30d6eda8a8ce867d
Studio               : 0.408.0
```

**Comment by janpio** [comment]
We need to confirm the described behavior in 2.22.0, reproduce the failure in recent versions and then understand how and why this changed - and then decide if this is something we can or want to undo, and how to properly document this.

**Comment by pantharshit00** [comment]
I think this was caused when we had a regression which made `env` to stop working with `binaryTargets`: https://github.com/prisma/prisma/issues/7295

I believe we should match the documented behaviour which is `PRISMA_BINARY_TARGET='native,darwin'` or we should define what we are expecting here.

**Comment by janpio** [comment]
> I believe we should match the documented behaviour

I don't think this is documented yet explicitly. The `env()` can replace values in the schema, and the expected value here is `["native", "darwin"]`. (`PRISMA_CLI_BINARY_TARGETS` is something different that affects the CLI, not the Client [yep, confusing but still true])

**Comment by gimtonic** [comment]
Dont work env variable if binaryTargets is not "native"
```json
  "devDependencies": {
    "prisma": "^3.13.0"
  },
 ```

**Comment by janpio** [comment]
That sounds unrelated @gimtonic. Please open a new issue and provide all information the issue template asks for.

**Comment by gimtonic** [comment]
Sorry. The problem is not relevant.
I fixed this by copying yhe .env file to inside docker container.

**Comment by lmanerich** [comment]
Hi there, this issue was still haunting prisma 4, so i figure out and workaround:

You need to use the env() method directly as the value and set the env to be the array

```
generator client {
  provider = "prisma-client-js"
  binaryTargets = env("PRISMA_BINARY_TARGET")
}
```

```
PRISMA_BINARY_TARGET=["native"]
```

Hope this helps

**Comment by svedova** [comment]
I believe it would be super nice if `Prisma` defaults to an environment variable (such as `PRISMA_BINARY_TARGET`) so that CI environments can set it without having to modify the `schema.prisma` file.

Shall I open a new issue for this?

**Comment by svedova** [comment]
> I believe it would be super nice if `Prisma` defaults to an environment variable (such as `PRISMA_BINARY_TARGET`) so that CI environments can set it without having to modify the `schema.prisma` file.
> 
> Shall I open a new issue for this?

I guess by providing the following environments, it looks like Prisma picks the correct binary target:

```
PRISMA_CLIENT_ENGINE_TYPE=binary
PRISMA_CLI_QUERY_ENGINE_TYPE=binary
PRISMA_CLI_BINARY_TARGETS=rhel-openssl-1.0.x
```

No need to update the configuration.

**Comment by janpio** [comment]
Yes please, being able to _override_ the configuration seems different enough to treat it via its own issue.

**Comment by svedova** [comment]
> Yes please @svedova, being able to _override_ the configuration seems different enough to treat it via its own issue. Thanks.

@janpio created: https://github.com/prisma/prisma/issues/20931
